components/brave_rewards/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -0,0 +1,91 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { types } from '../../constants/rewards_panel_types'
+import * as storage from '../storage'
+import { getTabData } from '../api/tabs_api'
+
+export const rewardsPanelReducer = (state: RewardsExtension.State | undefined, action: any) => {
+  if (state === undefined) {
+    state = storage.load()
+  }
+
+  const startingState = state
+
+  const payload = action.payload
+  switch (action.type) {
+    case types.CREATE_WALLET:
+      chrome.braveRewards.createWallet()
+      break
+    case types.ON_WALLET_CREATED:
+      state = { ...state }
+      state.walletCreated = true
+      break
+    case types.ON_WALLET_CREATE_FAILED:
+      state = { ...state }
+      state.walletCreateFailed = true
+      break
+    case types.ON_TAB_ID:
+      if (payload.tabId) {
+        getTabData(payload.tabId)
+      }
+      break
+    case types.ON_TAB_RETRIEVED:
+      const tab: chrome.tabs.Tab = payload.tab
+      if (
+        !tab ||
+        !tab.url ||
+        tab.incognito ||
```

Comment:
do tor windows return true for `tab.incognito`?

Reply:
correct